### PR TITLE
splitting into multiple project

### PR DESCRIPTION
**Pull request by kazvictor**
child of #61 

## Proposed Changes
* Splitting the code into multiple projects.

### Details
The idea behind the PR is to split the code into multiple projects:
 * The core project
 * The Gradle plugin

The reasoning for this is so that we could eventually reuse the core code to build other tools and utilities. For example, a command line application for use within the Gateway docker container to apply environment.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -20,62 +16,49 @@ tag {
     }
 }
 
-sourceCompatibility = 1.8
-targetCompatibility = 1.8
+subprojects {
+    apply plugin: 'java'
+    apply plugin: 'idea'
+    apply plugin: 'jacoco'
+    apply plugin: "com.srcclr.gradle"
```
Comment (yves-chan) [inline build.gradle]: I don't see the java, idea, and jacoco plugins defined in the subprojects, are they loaded by default?

Comment (kazvictor) [reply build.gradle]: Try running
`./gradlew gradle-plugin:publishToMavenLocal`

Comment (yves-chan) [reply build.gradle]: When I run `./gradlew gradle-plugin:publishToMavenLocal`                          

I get this error:
```                                                          
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=512m; support was removed in 8.0
Listening for transport dt_socket at address: 5005

FAILURE: Build failed with an exception.

* What went wrong:
Task 'publishToMavenLocal' not found in project ':gradle-plugin'.

* Try:
Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

Comment (joao-borges) [reply build.gradle]: Not working for me too

Comment (joao-borges) [reply build.gradle]: Fixed

**Pull request merged**